### PR TITLE
feat: register custom sanitizers

### LINKED ISSUE #14: Feature Request: Support custom sanitizers
Many struct-tag packages support the ability for consuming projects to inject custom directives into their struct tags. This functionality would greatly extend the flexibility and usability of this project. As a simple example, I need the ability to ensure a string starts with an uppercase letter. I'd like to leave all of the other characters in the string as-is.

For reference, here is another struct-tag project does it: [`httpin`](https://ggicci.github.io/httpin/directives/custom)

If this idea is approved, I'm willing to contribute a PR to implement it.

### PR DESCRIPTION
**Pull request by shawalli**
This PR adds support for custom sanitizers, which may be registered with a unique tag name. These custom sanitizers run before the built-in type sanitizers.

I think the "built-in" type sanitizers could be organized in a similar fashion. However, right now they operate in a de-centralized manner, where the actual sanitizer name is only known inside the type sanitizer.

Closes #14.

### CODE REVIEW

File: sanitize_test.go
Comment (mattwelke) [inline sanitize_test.go:77]: Can you add a test case that would show the expected result if the capfirst custom sanitizer function, as defined here, were used on a non-string field? For example, am `int` field?

**Review by mattwelke — CHANGES_REQUESTED** [review]

File: sanitize.go
Comment (hscasn) [inline sanitize.go:180]:
```suggestion
		for tag := range tags {
```

File: sanitize.go
Comment (hscasn) [inline sanitize.go:182]: (Just a comment)
I guess this is ok, since the default behaviour of the library is to ignore unknown tags. But now I wonder if we are doing the right thing by not even logging it as a potential problem. I'll give it more thought.

File: README.md
Comment (hscasn) [inline README.md:117]:
```suggestion
    Sanitizer: exclaim,
```

File: README.md
Comment (hscasn) [inline README.md:94]: Can you add a small comment describing what this sanitizer does?

File: README.md
Comment (hscasn) [inline README.md:119]: Can you add an example of a struct using this sanitizer, with a before and after? Something similar to what we have for the usage example:

```
func main() {
    d := Dog{
        Name: "Borky Borkins",
        Breed: nil,
    }

    s := sanitizer.New()
    s.Sanitize(&d)

    fmt.Printf("Name: %s, Breed: %s", d.Name, d.Breed)
    // Name: borky, Breed: unknown
}
```

**Review by hscasn — CHANGES_REQUESTED** [review]
Looks pretty good overall. Just a few requests.

File: sanitize.go
Comment (shawalli) [reply sanitize.go:182]: That makes sense. As it currently stands, the built-in tags exist in each type-sanitizer, so it's not possible for the custom sanitizer to know whether a tag directive is a custom sanitizer, a built-in sanitizer, or does not exist.

This could be fixed with a larger-scale overhaul of the type sanitizers by registering them in a similar fashion to the custom sanitizers, if that's a direction you'd like to move to over time.

Comment (hscasn) [reply sanitize.go:182]: I think this way is good enough. I think it's better to keep the behaviour consistent for both custom and standard sanitizers.

File: README.md
Comment (shawalli) [reply README.md:95]: Done

File: README.md
Comment (shawalli) [reply README.md:142]: Done

File: sanitize_test.go
Comment (shawalli) [reply sanitize_test.go:77]: Done

**Review by hscasn — APPROVED** [review]
LGTM. Please wait for a review from @mattwelke before merging

File: sanitize_test.go
Comment (mattwelke) [inline sanitize_test.go:142]: Awesome, this is exactly why I asked for the test to be added. I wanted to see what the design was going to be when a tag was used for a custom sanitizer for an invalid type.

**Review by mattwelke — APPROVED** [review]
LGTM 🚢 

@hscasn I think as it stands, we don't have automation set up to create a release automatically when a new commit hits the main branch. One of us would have to create a GitHub release manually. I think the previous commit hitting main is also pending release.

**Pull request merged**
